test/test_entanglement_tracker_grid.jl

```diff
@@ -0,0 +1,151 @@
+using Revise
+using QuantumSavory
+using ResumableFunctions
+using ConcurrentSim
+using QuantumSavory.ProtocolZoo
+using QuantumSavory.ProtocolZoo: EntanglementCounterpart, EntanglementHistory, EntanglementUpdateX, EntanglementUpdateZ
+using Graphs
+using Test
+
+if isinteractive()
+    using Logging
+    logger = ConsoleLogger(Logging.Debug; meta_formatter=(args...)->(:black,"",""))
+    global_logger(logger)
+    println("Logger set to debug")
+end
+
+##
+# Here we test entanglement tracker and swapper protocols on an arbitrary hardcoded path of long-range connection inside of what is otherwise a 2D grid
+# We do NOT test anything related to automatic routing on such a grid -- only the hardcoded path is tested
+##
+
+## Custom Predicates
+
+function top_left(net, node, x)
```

Comment:
I am waiting on CI and will then merge. However, this is an O(n) check for something that should be O(1).

From a node index get its x y coordinates (currently you get only one of them).
Then do the same for the other node.

Then just compare them to verify that you are to the left (one comparison) and to the top (second comparison).

No need for a for loop.

Submit a PR with the fix and feel free to merge it. It should also make this function much simpler.